common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java

```diff
@@ -0,0 +1,125 @@
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.internal.broker;
+
+import android.accounts.AccountManager;
+import android.accounts.AuthenticatorDescription;
+import android.content.Context;
+import android.content.Intent;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+/**
+ * Client that wraps the code necessary to bind to the MicrosoftAuthService (Android Bound Service)
+ */
+public class MicrosoftAuthClient {
+
+    private static final String TAG = MicrosoftAuthClient.class.getSimpleName();
+    private static final String MICROSOFT_AUTH_SERVICE_INTENT_FILTER = "com.microsoft.workaccount.brokeraccount.MicrosoftAuth";
+    private static final String MICROSOFT_AUTH_SERVICE_CLASS_NAME = "com.microsoft.workaccount.brokeraccount.MicrosoftAuthService";
+
+    private Context mContext;
+    private MicrosoftAuthServiceConnection mMicrosoftAuthServiceConnection;
+    private Intent mMicrosoftAuthServiceIntent;
+
+    /**
+     * Constructor for the Microsoft Auth Client
+     *
+     * @param context
+     */
+    public MicrosoftAuthClient(Context context) {
+        mContext = context;
+        mMicrosoftAuthServiceIntent = getIntentForAuthService(mContext);
+    }
+
+    /**
+     * Binds to the service and returns a future that provides the proxy for the calling the Microsoft auth service
+     *
+     * @return MicrosoftAuthServiceFuture
+     */
+    public MicrosoftAuthServiceFuture connect() {
+
+        MicrosoftAuthServiceFuture future = new MicrosoftAuthServiceFuture();
+        mMicrosoftAuthServiceConnection = new MicrosoftAuthServiceConnection(future);
+
+        final boolean serviceBound = mContext.bindService(mMicrosoftAuthServiceIntent, mMicrosoftAuthServiceConnection, Context.BIND_AUTO_CREATE);
+        Logger.verbose(TAG + "connect", "The status for MicrosoftAuthService bindService call is: " + Boolean.valueOf(serviceBound));
+
+        if (!serviceBound) {
+            disconnect();
+            throw new RuntimeException("Unable to bind to service");
```

Comment:
based on our current struggles with boundService, crashing the app here might be problematic, we should look at throwing a client exception saying broker not reponding

Reply:
This is work in progress relative to exceptions.  The handler upstream can convert this to whatever i like.

Reply:
I fixed this with clientException for now....